### DISCUSSION TITLE
Multi-Factor Authentication with Keycloak

### DISCUSSION THREAD

**Discussion by Bredstone** [keycloak/keycloak#16194] [Q&A]
Hello, hope you're doing well!!

I know that is possible to configure an authentication flow with multiple authentication methods configured as 2FA, allowing the user to choose between them. However, by default one of the authentication methods is presented with a `Try another` link and only then user can navigate to the 2FA selection screen.

I'd like to know if it is possible to configure authentication flow in a way that this selection screen is presented by default, before one of the authentication methods, so user can choose how it wants to proceed.

Should I be implementing my own custom SPI to do that or I am missing some Keycloak configuration??

Thanks in advance!!

**Comment by primeIndex** [comment]
Hey Bredstone, 
just wanted to check if you've found any solution for this?
If you have, please share it with me.

Thanks!
